storage/series/series.go

```diff
@@ -108,27 +117,35 @@ func (s *dbSeries) Tick() (TickResult, error) {
 	// In best case when explicitly asked to drain may have no
 	// stale buckets, cheaply check this case first with a Rlock
 	s.RLock()
+
 	needsDrain := s.buffer.NeedsDrain()
 	needsBlockUpdate := s.needsBlockUpdateWithLock()
 	r.ActiveBlocks = s.blocks.Len()
+
 	s.RUnlock()
 
 	if !needsDrain && !needsBlockUpdate {
 		return r, nil
 	}
 
 	s.Lock()
-	// Recalculate in case changed during lock promotion
-	needsDrain = s.buffer.NeedsDrain()
-	needsBlockUpdate = s.needsBlockUpdateWithLock()
+
+	// NB(r): don't bother to check if needs drain or needs
+	// block update still holds as running both these checks
+	// are relatively expensive and running these methods
+	// will be a no-op in case conditions no longer hold.
 	if needsDrain {
```

Comment:
This is fine, it's exactly the same as if this method was called a few nanoseconds earlier and needsDrain would've been false in both places.  This helped save a lot of CPU for ticking.  

Besides when checking when to "flush" or not at the end of a tick, the flush manager looks at the start time of the tick:
```go
// Begin ticking
start := d.nowFn()

// .. actually do the tick

if d.fsm.ShouldRun(start) {
  // actually flush
}
```